kore/src/Kore/Builtin/Map.hs

```diff
@@ -425,6 +433,25 @@ evalSize =
                 . Map.size
                 $ _map
 
+evalValues :: Builtin.Function
+evalValues =
+    Builtin.functionEvaluator evalValues0
+  where
+    evalValues0 _ resultSort = \arguments ->
+        Builtin.getAttemptedAxiom $ do
+            let _map =
+                    case arguments of
+                        [_map] -> _map
+                        _ -> Builtin.wrongArity Map.valuesKey
+            _map <- expectConcreteBuiltinMap Map.valuesKey _map
+            Builtin.List.returnList resultSort
+                . fmap TermLike.fromConcrete
+                . Seq.fromList
+                . catMaybes
+                . fmap (TermLike.asConcrete . Domain.getMapValue)
```

Comment:
BTW, why are you checking that the result is concrete?

Reply:
Good catch. There was some confusion with `getMapValue`, I'll remove the `asConcrete`/`fromConcrete` calls since they're not really needed.

Reply:
So `getMapValue` doesn't require a concrete map?

Reply:
I think the map is still concrete, see `_map <- expectConcreteBuiltinMap Map.lookupKey _map` above, but the _values_ shouldn't be forced to concrete here.